### PR TITLE
[test optimization] Update ESM note about mocha

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do? What is the motivation?

`dd-trace-js` has ESM support now, so there's no limitation to mocha's ES modules-first approach. 

### Merge instructions

Merge readiness:
- [x] Ready for merge

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### Additional notes
https://docs-staging.datadoghq.com/juan-fernandez/update-esm-support/tests/setup/javascript/?tab=ciproviderwithautoinstrumentationsupport#mocha-es-modules

### CODE REVIEW

File: content/en/tests/setup/javascript.md [hunk 1]
```diff
@@ -126,8 +126,14 @@ Just like tags, you can add custom measures to your tests by using the current a
 
 For more information about custom measures, see the [Add Custom Measures Guide][2].
 
+### Mocha ES modules
```
[the remark this answers is not in the document]
Comment (juan-fernandez) [reply content/en/tests/setup/javascript.md:129]: shouldn't it be `### Mocha ECMAScript modules (ESM)` ? following nomenclature here: https://nodejs.org/api/esm.html

[the remark this answers is not in the document]
Comment (neko-dd) [reply content/en/tests/setup/javascript.md:129]: I was going by what was in https://github.com/datadog/dd-trace-js?tab=readme-ov-file#ecmascript-modules-esm-support, but I think you're right that your link is more authoritative.

**Review by anmarchenko — APPROVED** [review]

**Review by anmarchenko — APPROVED** [review]

**Pull request merged**